components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -130,8 +130,10 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             }
         }
 
-        String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL();
-        String retryPage = ConfigurationFacade.getInstance().getAuthenticationEndpointRetryURL();
+        ConfigurationFacade configurationFacade = ConfigurationFacade.getInstance();
+        String loginPage = IdentityUtil.getServerURL(configurationFacade.getAuthenticationEndpointURL(), false, false);
+        String retryPage = IdentityUtil.getServerURL(configurationFacade.getAuthenticationEndpointRetryURL(), false,
+                false);
```

Comment:
Looks like we should be fixing at 'getAuthenticationEndpointURL' method of 'ConfigurationFacade' not at basic authenticator